### PR TITLE
chore: prepare for reusable workflows

### PR DESCRIPTION
**Pull request by lukasmrtvy**
Other part of PR https://github.com/apify/apify-docs-private/pull/31

### CODE REVIEW

**Review by mtrunkat — APPROVED** [review]
Nice

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    - 'nginx.conf'
+
+jobs:
+  deploy:
+      name: 'Deploy'
+      runs-on: ubuntu-latest
+      steps:
+        - run: gh workflow run deploy.yaml --repo apify/apify-docs-private
+          env:
+            GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
```
Comment (fnesveda) [inline .github/workflows/deploy.yaml:15]:
```suggestion
            GITHUB_TOKEN: ${{ secrets.APIFY_SERVICE_ACCOUNT_GITHUB_TOKEN }}
```

File: .github/workflows/invalidate.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  workflow_dispatch:
+
+jobs:
+  invalidate:
+      name: 'Invalidate'
+      runs-on: ubuntu-latest
+      steps:
+        - run: gh workflow run invalidate.yaml --repo apify/apify-docs-private
+          env:
+            GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
```
Comment (fnesveda) [inline .github/workflows/invalidate.yaml:13]:
```suggestion
            GITHUB_TOKEN: ${{ secrets.APIFY_SERVICE_ACCOUNT_GITHUB_TOKEN }}
```

**Review by fnesveda — CHANGES_REQUESTED** [review]
Looks good, let's just use the GitHub token of the Apify Service Account rather than the one from this repo. It should have enough permissions for this.

**Review by fnesveda — APPROVED** [review]

**Pull request merged**